dcos_tunnel/cli.py

```diff
@@ -632,37 +654,49 @@ def _vpn(port, config_file, user, privileged, ssh_port, host,
 
         msg = "\nWaiting for VPN server in container '{}' to come up..."
         emitter.publish(msg.format(container_name))
+
         scom = ("until "
-                """[ "$(docker inspect --format='{{ .State.Running }}' """
+                """[ "$(%s inspect --format='{{ .State.Running }}' """
                 """%s 2>/dev/null)" = "true" ] 2>/dev/null; do sleep 0.5; """
-                """done""") % (container_name)
-        scom += (";"
-                 """docker exec {} "sh -c 'until [ -s {} ]; do sleep 0.5; """
-                 """done'" """).format(container_name, remote_keyfile)
-        scom += (";"
-                 """docker exec {} "sh -c 'until [ -s {} ]; do sleep 0.5; """
-                 """done'" """).format(container_name, remote_clientfile)
+                """done""") % (docker_cmd, container_name)
+        scom += (" && "
+                 """{} exec {} sh -c 'until [ -s {} ]; do sleep 0.5; """
+                 """done' """).format(docker_cmd, container_name,
+                                      remote_keyfile)
+        scom += (" && "
+                 """{} exec {} sh -c 'until [ -s {} ]; do sleep 0.5; """
+                 """done' """).format(docker_cmd, container_name,
+                                      remote_clientfile)
         server_setup = client.exec_command(scom, get_pty=True)
-        ssh_exec_wait(server_setup)
 
-        scom = ("""docker inspect --format='"""
+        if not vpn_server.is_alive():
```

Comment:
Sure, but what I'm getting at is that if the VPN server crashes early on for some reason than with this check we never get stuck in the until loop and it's virtually free.